### PR TITLE
#2840: Updated gallery cards for label types that do not support severity ratings

### LINKED ISSUE #2840: Gray out severity levels for unsupported label types
Now that we've removed severity ratings from some label types, we need to decide how to show (or hide) this info in other visualizations like Sidewalk Gallery. 

For example, here's Pedestrian Signal before our recent [change](https://github.com/ProjectSidewalk/SidewalkWebpage/pull/2838):
image

And here's what it looks like after:
image

I think I might argue, however, that for visual consistency with other gallery cards, we shouldn't make the severity ratings fully invisible but rather "disabled" by graying them out.

image

### PR DESCRIPTION
**Pull request by GPA0VERKD**
Resolves #2840

Changed Gallery cards with unsupported severity ratings to have grayed out severity ratings, as well as a tooltip that, wehn hovered over, explains why they have no rating.

##### Before/After screenshots (if applicable)
Before:
image

After:
image

##### Testing instructions
1. Go to the gallery tab
2. Under the "Show" drop down on the left, hit "Pedestrian Signal," or any label type that does not support severity ratings. (if the server you're working on doesn't have any pedestrian signals tagged, go into the audit tab and throw in a couple random pedestrian signal labels, not in production server ofc!!!)
3. Check that the Severity Rating is grayed out on all cards, and that a tooltip is there. When hovered over, a message box should appear.
4. Check in all languages!

##### Things to check before submitting the PR 
- [x] I've written a descriptive PR title.
- [x] I've added/updated comments for large or confusing blocks of code.
- [x] I've included before/after screenshots above.
- [x] I've asked for and included translations for any user facing text that was added or modified.

### CODE REVIEW

**Comment by misaugstad** [comment]
I think that we can skip out on the '?' icon and just have the tooltip on the grayed out section. Do you have this on the expanded view when you click on a card as well?

**Comment by GPA0VERKD** [comment]
Since we dont have a grayed out image for the smiley faces in the expanded view I just made them grayed out circles when there's no severity rating, does that work?

**Comment by misaugstad** [comment]
There's this CSS property called `filter` that you can apply to an image. I just tested out setting `filter: opactity(0.5)` to one of the smileys and it looks like it's grayed out :ok_hand: 
Screenshot from 2022-07-05 12-33-26

**Comment by misaugstad** [comment]
I think that what you did is make it so that whenever a label doesn't have severity, you show the "unsupported" tooltip. But we actually only wanted to show this tooltip for the two label types where you aren't even able to add a severity rating: "Can't see the sidewalk" and "Pedestrian signal".

This does make me realize that we should do the grayed out with a tooltip for other label types if severity isn't provided as well. But that tooltip should just say that the user did not provide a severity rating.

In addition, the Can't See the Sidewalk label type is still showing no severity rating at all!

Here's screenshots with the various bits of incorrect behavior:
Screenshot from 2022-07-06 13-01-40
Screenshot from 2022-07-06 13-03-28
Screenshot from 2022-07-06 13-02-09

**Comment by misaugstad** [comment]
Here's the one thing from the list above that I think you missed!

>This does make me realize that we should do the grayed out with a tooltip for other label types if severity isn't provided as well. But that tooltip should just say that the user did not provide a severity rating.

**Comment by misaugstad** [comment]
A few more notes!
1. I made an update myself to remove old severity ratings for Pedestrian Signal labels since they were allowed for a couple weeks there. The existence of that data was causing bugs here, and I figured that wasn't worth it. So I cleaned up my old mess :)
2. Can you change this text slightly to say "A severity rating was not provided for this label"
    Screenshot from 2022-07-19 13-52-35
4. The expanded view tooltip is off center
    Screenshot from 2022-07-19 12-38-50
6. The grayed out text looks really bad on validated labels. This is because you're changing the font color to lightgray. You should probably just be keeping it as black and reducing the opacity.
    Screenshot from 2022-07-19 13-52-40

Those last two should have become clear with extensive testing, so I challenge you to step up your code testing game in future iterations and PRs! ;)

**Comment by GPA0VERKD** [comment]
@misaugstad Ready for another review!

File: public/javascripts/Gallery/src/displays/SeverityDisplay.js [hunk 1]
```diff
@@ -7,11 +7,17 @@
  * @param {Boolean} isModal a toggle to determine if this SeverityDisplay is in a modal, or in a card
  * @returns {SeverityDisplay} the generated object
  */
-function SeverityDisplay(container, severity, isModal=false) {
+function SeverityDisplay(container, severity, labelType, isModal=false) {
     let self = this;
     self.severity = severity;
     self.severityContainer = container;
 
+    // list of label types where severity ratings are not supported
```
Comment (misaugstad) [inline public/javascripts/Gallery/src/displays/SeverityDisplay.js:15]: Can you make the small edits to the comments to fit the [style guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-Guide)?

File: public/javascripts/Gallery/src/displays/SeverityDisplay.js [hunk 2]
```diff
@@ -26,36 +32,62 @@ function SeverityDisplay(container, severity, isModal=false) {
         if (isModal) {
             // Add bold weight. Find better way to do this.
             title.classList.add('modal-severity-header');
+            // centers tooltip
+            holder.classList.add('modal-no-severity-box')
```
Comment (misaugstad) [inline public/javascripts/Gallery/src/displays/SeverityDisplay.js:36]: It looks like this CSS class isn't specific to there being no severity (even though that's what's motivating the change), so maybe change the name to something that isn't specific to a lack of severity?

File: public/javascripts/Gallery/src/displays/SeverityDisplay.js [hunk 3]
```diff
@@ -26,36 +32,62 @@ function SeverityDisplay(container, severity, isModal=false) {
...
-            circles.push(severityCircle);
-        }
-
-        // Highlight the correct severity.
-        // We do so by darkening a number of circles from the left equal to the severity. For example, if the severity
-        // is 3, we will darken the left 3 circles.
-        // TODO: rename these once confirmed. also, we can probably move this to the upper loop.
-        if (severity) {
-            for (let i = 0; i < severity; i++) {
+            if (unsupported || !severity) {
```
Comment (misaugstad) [inline public/javascripts/Gallery/src/displays/SeverityDisplay.js:53]: Can we switch to checking specifically if `severity` is null or undefined (not sure which it is in this case). I think I'd prefer that over checking if it is falsey

File: public/javascripts/Gallery/src/displays/SeverityDisplay.js [hunk 4]
```diff
@@ -26,36 +32,62 @@ function SeverityDisplay(container, severity, isModal=false) {
...
+            circles.push(severityCircle);
+        }
+
+        if (!severity) {
+            // add tooltip if no severity level
+            holder.setAttribute('data-toggle', 'tooltip');
+            holder.setAttribute('data-placement', 'top');
+
+            // change tooltip message depending on if the label is unsupported or user did not add severity rating
+            if (unsupported){
```
Comment (misaugstad) [inline public/javascripts/Gallery/src/displays/SeverityDisplay.js:85]: missing that whitespace between the `)` and `{` ;)

**Review by misaugstad — CHANGES_REQUESTED** [review]
Just added some super minor style changes. Everything is working great though! Just about ready to merge!

**Comment by GPA0VERKD** [comment]
Ready for another check!

**Review by misaugstad — APPROVED** [review]
Looks good, thanks @GPA0VERKD :)

**Pull request merged**